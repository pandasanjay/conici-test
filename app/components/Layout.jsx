import React from "react";
import Header from "./Header";
import Footer from "./Footer";

const Layout = ({ children }) => {
  return (
    <div>
      <Header title="Smart Check-out" />
      <div className="container">{children}</div>
      <Footer copywrite="@ 2018 - Hotel Beacons GmbH" />
    </div>
  );
};

export default Layout;
