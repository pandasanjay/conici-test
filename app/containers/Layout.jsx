import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import Header from "../components/Header";
import Footer from "../components/Footer";
import { connect } from "react-redux";
import { globalContentAction } from "../actions";
export class Layout extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.unlisten = this.props.history.listen(this.handelRouterEnter);
    this.handelRouterEnter();
  }
  componentWillUnmount() {
    this.unlisten();
  }
  handelRouterEnter(location, action) {
    this.props.globalContentAction();
  }
  render() {
    const title =
      this.props.globalReducer.pageContents &&
      this.props.globalReducer.pageContents.global.title;
    const copyright =
      this.props.globalReducer.pageContents &&
      this.props.globalReducer.pageContents.global.copyright;
    return (
      <div>
        <Header title={title} />
        <div className="container">{this.props.children}</div>
        <Footer copyright={copyright} />
      </div>
    );
  }
}
/*
 * mapDispatchToProps
 */
const mapDispatchToProps = dispatch => ({
  globalContentAction: () => dispatch(globalContentAction())
});

/*
 * mapStateToProps
 */
const mapStateToProps = state => ({
  ...state
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Layout));
