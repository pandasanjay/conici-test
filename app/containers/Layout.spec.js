import React from "react";
import { Layout } from "./Layout";
import { shallow } from "enzyme";
describe("Layout conatiner", () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <Layout
        globalReducer={{}}
        history={{ listen: jest.fn() }}
        globalContentAction={jest.fn()}
      />
    );
  });
  it("should have match with snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
