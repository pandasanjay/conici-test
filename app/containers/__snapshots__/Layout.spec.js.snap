// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout conatiner should have match with snapshot 1`] = `
<div>
  <Header />
  <div
    className="container"
  />
  <Footer />
</div>
`;
